pkg/crc/machine/bundle/metadata.go

```diff
@@ -128,7 +130,10 @@ func ExtractFromExecutable(sourcepath string) (*CrcBundleInfo, error) {
 		return nil, err
 	}
 
-	return GetCachedBundleInfo(filepath.Base(sourcepath))
+	if err := writeExtractionDone(bundleName); err != nil {
+		return nil, err
+	}
+	return GetCachedBundleInfo(bundleName)
 }
```

Comment:
Fwiw, the code duplication between `ExtractFromExecutable` and `Extract` seems to be avoidable if you add an `extractFromReader` helper, and open the file in `Extract` rather than opening it in `extract.Uncompress`